File: leetcode/python/49.py
import sys
sys.stdin = open('input.txt', 'r')
strs = sys.stdin.readline().strip().split()

counts = [[0 for _ in range(26)] for _ in range(len(strs))]
a_ord = ord('a')
d = dict()
for i in range(len(strs)):
    for s in strs[i]:
        counts[i][ord(s)-a_ord] += 1

for i in range(len(counts)):
    k = ''.join(map(str, counts[i]))
    if k in d:
        d[k].append(strs[i])
    else:
        d[k] = [strs[i]]

answer = []
for k in d:
    answer.append(d[k])

answer.sort()

print(answer)

File: leetcode/cpp/49.cpp
#include <bits/stdc++.h>

using namespace std;

vector<vector<string>> groupAnagrams(vector<string> &strs)
{
    unordered_map<string, vector<string>> mp;

    for (auto x : strs)
    {
        string word = x;
        sort(word.begin(), word.end());
        mp[word].push_back(x);
    }

    vector<vector<string>> ans;
    for (auto x : mp)
    {
        ans.push_back(x.second);
    }
    return ans;
}

void printAns(vector<vector<string>> ans)
{
    for (int i = 0; i < ans.size(); i++)
    {
        for (int j = 0; j < ans[i].size(); j++)
        {
            cout << ans[i][j] << " " << endl;
        }
        cout << endl;
    }
    cout << endl;
}

int main()
{
    freopen("input.txt", "r", stdin);
    int n;
    cin >> n;
    vector<string> strs;
    for (int i = 0; i < n; i++)
    {
        string tempStr;
        cin >> tempStr;
        strs.push_back(tempStr);
    }
    printAns(groupAnagrams(strs));
    return 0;
}